docker/checkout.bash

```diff
@@ -34,6 +34,10 @@ while [[ $# -gt 0 ]]; do
                   shift
                   shift
             ;;
+            -d)  # This is used to ignore -d arg from build-image.sh to avoid conflict with -b arg used in dockerfile.
```

Comment:
Will the conflict occur only if a `-d` arg is provided from the `Dockerfile`? If so, then I'd be okay with us removing this logic here, since it seems arbitrary to have a special case for `-d` and not other invalid arguments. Let me know if I'm misunderstanding this though

Reply:
I added this because when I was executing build-imae.sh with -d flag to build on develop but I was getting the build error as below:

root@sai-VirtualBox:/home/sai/stolrepos/carma-msgs/docker# ./build-image.sh -d
##### carma-msgs Docker Image Build Script #####
Building docker image for carma-msgs version: develop
Final image name: usdotfhwastoldev/carma-msgs:develop
[+] Building 82.8s (10/12)                                                                                                                                docker:default
 => [internal] load .dockerignore                                                                                                                                   0.2s
 => => transferring context: 2B                                                                                                                                     0.0s
 => [internal] load build definition from Dockerfile                                                                                                                0.3s
 => => transferring dockerfile: 1.31kB                                                                                                                              0.1s
 => [internal] load metadata for docker.io/usdotfhwastoldev/carma-base:develop                                                                                      0.0s
 => CACHED [1/8] FROM docker.io/usdotfhwastoldev/carma-base:develop                                                                                                 0.0s
 => [internal] load build context                                                                                                                                   1.9s
 => => transferring context: 67.32kB                                                                                                                                1.2s
 => [2/8] RUN  mkdir -p ~/.base-image/ros1_msgs_ws/src/carma_msgs                                                                                                   3.6s
 => [3/8] RUN  mkdir -p /home/carma/src                                                                                                                             3.5s
 => [4/8] COPY . /home/carma/.base-image/ros1_msgs_ws/src/carma_msgs/                                                                                              11.4s
 => [5/8] RUN  mkdir -p ~/.base-image/ros2_msgs_ws/src/carma_msgs                                                                                                   4.0s
 => [6/8] COPY . /home/carma/.base-image/ros2_msgs_ws/src/carma_msgs/                                                                                               2.4s
 => [7/8] RUN /home/carma/.base-image/ros1_msgs_ws/src/carma_msgs/docker/checkout.bash -d -b develop -r /home/carma/.base-image/workspace                          57.1s
 => => # + case $arg in                                                                                                                                                 
 => => # + [[ 5 -gt 0 ]]                                                                                                                                                
 => => # + arg=-d                                                                                                                                                       
 => => # + case $arg in                                                                                                                                                 
 => => # + [[                                                                                                                                                           
 => => # [output clipped, log limit 200KiB/s reached]

Reply:
Makes sense. I think we'll eventually refactor the `build-image.sh` scripts to only support incremental builds (i.e. support of the `--ros1` and `--ros2` args), which will also help avoid this issue.